decidim-proposals/app/controllers/decidim/proposals/collaborative_drafts_controller.rb

```diff
@@ -0,0 +1,220 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Proposals
+    # Exposes Collaborative Drafts resource so users can view and create them.
+    class CollaborativeDraftsController < Decidim::Proposals::ApplicationController
+      helper Decidim::WidgetUrlsHelper
+      helper ProposalWizardHelper
+      helper TooltipHelper
+      include FormFactory
+      include FilterResource
+      include CollaborativeOrderable
+      include Paginable
+
+      helper_method :geocoded_collaborative_draft
+      before_action :collaborative_drafts_enabled?
+      before_action :authenticate_user!, only: [:new, :create, :complete]
+      before_action :retrieve_collaborative_draft, only: [:show, :edit, :update, :request_access, :request_accept, :request_reject, :withdraw, :publish]
+
+      def index
+        @collaborative_drafts = search
+                                .results
+                                .includes(:authors)
+                                .includes(:category)
+                                .includes(:scope)
+
+        @collaborative_drafts = paginate(@collaborative_drafts)
+        @collaborative_drafts = reorder(@collaborative_drafts)
+      end
+
+      def show
+        @report_form = form(Decidim::ReportForm).from_params(reason: "spam")
+      end
+
+      def new
+        enforce_permission_to :create, :collaborative_draft
+        @step = :step_1
+
+        @form = form(CollaborativeDraftForm).from_params(
+          attachment: form(AttachmentForm).from_params({})
+        )
+      end
+
+      def compare
+        @step = :step_2
+        @similar_collaborative_drafts ||= Decidim::Proposals::SimilarCollaborativeDrafts
```

Comment:
This only looks for Collaborative drafts, shouldn't it also look for similar proposals? (And viceversa)

I feel like there's a lot of duplicated code in this PR. I know I said I was OK with adding a new entity (the Collaborative draft), but maybe a flag should have been raised while developing this PR to rethink the approach.

A lot of logic wouldn't have needed to be duplicated if a proposal could have a `collaborative_draft` state.

Reply:
Yes, this has been discussed in #3079